lib/net/ssh/buffer.rb

```diff
@@ -347,7 +347,7 @@ def write(*data)
       # Optimized version of write where the caller gives up ownership of string
       # to the method. This way we can mutate the string.
       def write_moved(string)
-        @content << string.force_encoding('BINARY')
+        @content << string.dup.force_encoding('BINARY')
```

Comment:
see comment above the method, we don't want `dup` here for sure

Reply:
Duh, sorry. My bad

Reply:
I feel like it should fail a test then.